snmp/tests/test_profiles.py

```diff
@@ -48,6 +48,25 @@
 pytestmark = pytest.mark.usefixtures("dd_environment")
 
 
+def _verify_profile(check, profile, name):
+    __tracebackhide__ = True
+    message = None
+    try:
+        check._config.refresh_with_profile(profile)
+    except ConfigurationError as e:
+        message = "Profile `{}` is not configured correctly: {}".format(name, e)
+    return message
+
+
+def test_load_profiles(aggregator):
+    instance = common.generate_instance_config([])
+    check = SnmpCheck('snmp', {}, [instance])
+    for name, profile in check.profiles.items():
+        profile_error = _verify_profile(check, profile, name)
+        if profile_error:
+            pytest.fail(profile_error)
```

Comment:
Hey, after sleeping on it I think it might be better to just stick with the duplicate error messages…

The usage induced by `__tracebackhide__` is actually quite exotic and most likely unknown to most people, so we should probably stick to an as-simple-as-possible setup (which also has the pro of showing the exact location where the parser has failed; useful if there's a problem in the parser itself). WDYT?

```suggestion
        try:
            check._config.refresh_with_profile(profile)
        except ConfigurationError as e:
            pytest.fail("Profile `{}` is not configured correctly: {}".format(name, e))
```

Sorry for the bikeshedding…

Reply:
I agree, the extra information is probably more helpful in this case.